### PR TITLE
Feat/718 integrate node validation

### LINKED ISSUE #718: Integrate node switcher to initial node selection
Open the node switcher when loading the initial node fails.

Relies the implementation from: https://github.com/vegaprotocol/frontend-monorepo/issues/622

### PR DESCRIPTION
**Pull request by notbot00**
# Related issues 🔗

Closes #718 

# Description ℹ️

- adds custom node support to the node switcher dialog
- integrates the node switcher dialog with initial error messages to show when the default node selection does not succeed

# Demo 📺

Screenshot 2022-07-14 at 08 55 36
Screenshot 2022-07-14 at 08 56 15
Screenshot 2022-07-14 at 08 57 12
Screenshot 2022-07-14 at 08 58 50
Screenshot 2022-07-14 at 09 07 50
Screenshot 2022-07-14 at 09 07 26

MOBILE (based on https://github.com/vegaprotocol/frontend-monorepo/issues/622#issuecomment-1167226655):

Screenshot 2022-07-14 at 11 10 48

# Technical 👨‍🔧

- VALID NODE = 
    - it is a valid url (checked by initialising a `new URL`)
    - returns blockHeight and chainId (query statistics)
    - it's on the same network as the app (chainId does include VEGA_ENV from the environment)
    - does have ssl working (checked by one-time subscription to busEvents change)
    
- open the node switcher on startup:
    - there is no VEGA_URL specified in the environment AND:
         - cannot load the network config
         - network config is not valid
         - none of the nodes in the config is a VALID NODE
         - the selected node turns out not to have ssl support (detached ssl validation due to high latency on some networks)
    - there is a VEGA_URL specified in the environment AND:
        - it is not a VALID NODE
        - turns out not to have ssl support (detached ssl validation due to high latency on some networks)
- open the node switcher on user interaction:
    - when clicking the Edit link in the footer network info (token and explorer apps)
    
- when selecting a node:
    - from the default network nodes:
        - only the VALID NODES can be selected
    - a custom one:
        - has to be a VALID NODE
         
Screenshot 2022-07-14 at 08 29 08

### CODE REVIEW

**Review by mattrussell36 — APPROVED** [review]

**Review by MadalinaRaicu — APPROVED** [review]

File: libs/environment/src/hooks/use-nodes.spec.tsx [hunk 1]
```diff
@@ -0,0 +1,402 @@
...
+  ]);
+
+beforeEach(() => {
+  // @ts-ignore allow adding a mock return value to mocked module
+  createClient.mockImplementation(() => createMockClient());
+});
+
+afterAll(() => {
+  // @ts-ignore allow deleting the spy function after we're done with the tests
+  delete window.performance.getEntriesByName;
```
Comment (dexturr) [inline libs/environment/src/hooks/use-nodes.spec.tsx:60]: question: can we not use https://jestjs.io/docs/jest-object#jestclearallmocks?

File: libs/environment/src/hooks/use-nodes.spec.tsx [hunk 2]
```diff
@@ -0,0 +1,402 @@
...
+      name: url,
+      startTime: 0,
+      toJSON: () => ({}),
+      duration: MOCK_DURATION,
+    },
+  ]);
+
+beforeEach(() => {
+  // @ts-ignore allow adding a mock return value to mocked module
+  createClient.mockImplementation(() => createMockClient());
```
Comment (dexturr) [inline libs/environment/src/hooks/use-nodes.spec.tsx:55]: question: why is this required? is it not possible for us to use MockedProvider like in other tests?

File: libs/environment/src/hooks/use-environment.tsx [hunk 3]
```diff
@@ -42,14 +102,15 @@ export const EnvironmentProvider = ({
...
+        networkError,
         setNodeSwitcherOpen: () => setNodeSwitcherOpen(true),
       }}
     >
-      {config && (
+      {config && isNodeSwitcherOpen && (
         <NodeSwitcherDialog
-          dialogOpen={isNodeSwitcherOpen}
-          toggleDialogOpen={setNodeSwitcherOpen}
+          dialogOpen={true}
```
Comment (dexturr) [inline libs/environment/src/hooks/use-environment.tsx:111]: question: why do we now have to set this to true?

**Review by dexturr — APPROVED** [review]

File: libs/environment/src/hooks/use-environment.tsx (hunk 3 above)
Comment (notbot00) [reply libs/environment/src/hooks/use-environment.tsx]: true, this was from some leftover refactor. moved things around a bit to make this better. thanks!
